ddtrace/tracer/rules_sampler.go

```diff
@@ -87,6 +87,9 @@ type SamplingRule struct {
 
 // match returns true when the span's details match all the expected values in the rule.
 func (sr *SamplingRule) match(s *span) bool {
+	if sr.TargetRoot && s.root().SpanID != s.SpanID {
```

Comment:
How likely is `s.root()` to return `nil`, or for `s` to be `nil`?

Reply:
Span can be newly created with all the necessary / fields / context filled out, but never nil. Subsequently, root wouldn't be nil. The match function is called only within sampling functionality on a trace / chunk, and thus span is never empty.